packages/realm_dart/lib/src/logging.dart

```diff
@@ -0,0 +1,182 @@
+// Copyright 2024 MongoDB, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'dart:async';
+
+import 'package:logging/logging.dart';
+
+import 'native/realm_core.dart';
+
+// Using classes to make a fancy hierarchical enum
+sealed class RealmLogCategory {
```

Comment:
Can we use `LogCategory` here? I know we do sometimes prefix stuff with `Realm`, but I'd prefer if we only do that for things that go in a Realm (database) and not for other API.

Reply:
We could. 

I'm am/was little hesitant to call `RealmLogRecord` for `LogRecord` as that is the same name the [logging](https://pub.dev/packages/logging) package use, and I foresee that people will want to connect with that package.

And once I started using a `Realm` prefix on that it felt more natural to use it throughout the interface - including here.

Obviously any name conflict can always be resolved with a simple `import x show y as z`, but since we are using the `Realm` prefix so many other places I added it for consistency.

So, it was a conscious choice, but I don't have strong opinions. Now you got the context, I'll let you do the call 😄

Reply:
I see - I don't have a strong opinion there, it's just a little annoying that we have to use a worse API for our stuff to avoid conflicting with another package, but I guess that's the price we pay for not being first-party.